photutils/psf/photometry.py

```diff
@@ -407,3 +355,361 @@ def _model_params2table(self, fit_model, star_group):
+    group_maker : callable or `~photutils.psf.GroupStarsBase`
+        ``group_maker`` should be able to decide whether a given star
+        overlaps with any other and label them as beloging to the same
+        group.  ``group_maker`` receives as input an
+        `~astropy.table.Table` object with columns named as ``id``,
+        ``x_0``, ``y_0``, in which ``x_0`` and ``y_0`` have the same
+        meaning of ``xcentroid`` and ``ycentroid``.  This callable must
+        return an `~astropy.table.Table` with columns ``id``, ``x_0``,
+        ``y_0``, and ``group_id``. The column ``group_id`` should cotain
+        integers starting from ``1`` that indicate which group a given
+        source belongs to. See, e.g., `~photutils.psf.DAOGroup`.
+    bkg_estimator : callable, instance of any `~photutils.BackgroundBase` subclass, or None
+        ``bkg_estimator`` should be able to compute either a scalar
+        background or a 2D background of a given 2D image. See, e.g.,
+        `~photutils.background.MedianBackground`.  If None, no
+        background subtraction is performed.
+    psf_model : `astropy.modeling.Fittable2DModel` instance
+        PSF or PRF model to fit the data. Could be one of the models in
+        this package like `~photutils.psf.sandbox.DiscretePRF`,
+        `~photutils.psf.IntegratedGaussianPRF`, or any other suitable 2D
+        model.  This object needs to identify three parameters (position
+        of center in x and y coordinates and the flux) in order to set
+        them to suitable starting values for each fit. The names of
+        these parameters should be given as ``x_0``, ``y_0`` and
+        ``flux``.  `~photutils.psf.prepare_psf_model` can be used to
+        prepare any 2D model to match this assumption.
+    fitshape : int or length-2 array-like
+        Rectangular shape around the center of a star which will be used
+        to collect the data to do the fitting. Can be an integer to be
+        the same along both axes. E.g., 5 is the same as (5, 5), which
+        means to fit only at the following relative pixel positions:
+        [-2, -1, 0, 1, 2].  Each element of ``fitshape`` must be an odd
+        number.
+    finder : callable or instance of any `~photutils.detection.StarFinderBase` subclasses
+        ``finder`` should be able to identify stars, i.e. compute a
+        rough estimate of the centroids, in a given 2D image.
+        ``finder`` receives as input a 2D image and returns an
+        `~astropy.table.Table` object which contains columns with names:
+        ``id``, ``xcentroid``, ``ycentroid``, and ``flux``. In which
+        ``id`` is an integer-valued column starting from ``1``,
+        ``xcentroid`` and ``ycentroid`` are center position estimates of
+        the sources and ``flux`` contains flux estimates of the sources.
+        See, e.g., `~photutils.detection.DAOStarFinder`.  If ``finder``
+        is ``None``, initial guesses for positions of objects must be
```

Comment:
finder *cannot* be none in this case, right? It looks to me like this class raises errors in that case (which is what was intended)